### PR TITLE
consensus: integrate dBFT pre-0.2.0

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Try out unreleased version to see how the integration is going.

### CODE REVIEW

**Comment by AnnaShaleva** [comment]
Unfortunately, transactions conversion can't be handled without new list allocation.

File: pkg/core/block/header.go [hunk 1]
```diff
@@ -13,6 +13,9 @@ import (
 	"github.com/nspcc-dev/neo-go/pkg/util"
 )
 
+// Version is a default Neo block version.
+const Version uint32 = 0
```
Comment (roman-khimov) [inline pkg/core/block/header.go:17]: We gonna have version 1 soon, so what `block.Version` would mean?

Comment (AnnaShaleva) [reply pkg/core/block/header.go:17]: I think we inevitably need constants for version, going with literals seems not good, especially with the upcoming version 1. How about adding `Version0` and `Version1` constants? Or we may even create a special type for version. What would you prefer?

Comment (roman-khimov) [reply pkg/core/block/header.go:17]: I think they'll be tied to hardforks. We don't yet have the next name, but current one can be `VersionInitial`. Then it'll be like `VersionDsmth`.

Comment (AnnaShaleva) [reply pkg/core/block/header.go]: Fixed.

File: pkg/core/block/header.go [hunk 2]
```diff
@@ -13,6 +13,9 @@ import (
 	"github.com/nspcc-dev/neo-go/pkg/util"
 )
 
+// VersionInitial is a default Neo block version.
```
Comment (roman-khimov) [inline pkg/core/block/header.go:16]: the default?

**Review by roman-khimov — APPROVED** [review]

File: pkg/core/block/header.go (hunk 2 above)
Comment (AnnaShaleva) [reply pkg/core/block/header.go]: Fixed.

**Pull request merged**